piff/polynomial_interp.py

```diff
@@ -0,0 +1,361 @@
+                                find the order of the parameter from self.
+        :param C:          A 2D matrix of polynomial coefficients in the form that
+                           the numpy polynomial form is expecting:
+                           p(x,y,c) = sum_{i,j} c_{ij} x^i y^j
+        :returns coeffs:    A 1D numpy array of coefficients of length self.nvariable
+        """
+        coeffs = numpy.zeros(self.nvariables[parameter_index])
+        for k,(i,j) in enumerate(self.indices[parameter_index]):
+            coeffs[k] = C[i,j]
+        return coeffs
+
+
+    def unpack_coefficients(self, parameter_index, coeffs):
+        """Unpack a sequence of parameters into the 2D matrix for the 
+        given parameter_index (which determines the order of the matrix)
+
+        This function is the inverse of pack_coefficients
+                           
+        :param parameter_index: The integer index of the parameter being used
+        :param coeffs:     A 1D numpy array of coefficients  of length self.nvariable
+        :returns:          A 2D matrix of polynomial coefficients in the form that
+                           the numpy polynomial form is expecting:
+                           p(x,y,c) = sum_{i,j} c_{ij} x^i y^j
+
+        """
+        k=0
+        n=self.orders[parameter_index]+1
+        C = numpy.zeros((n, n))
+        for k,(i,j) in enumerate(self.indices[parameter_index]):
+                C[i,j] = coeffs[k]
+                k+=1
+        return C
+
+    
+    def interpolationModel(self, pos, C):
+        """Generate the polynomial variation of some quantity at x and y
+        coordinates for a given coefficient matrix.
+
+        TODO: At the moment this function is expecting a numpy array of
+        shape (2,nstar) for the positions. We might want to use a galsim
+        position object instead since I think some code elsewhere in Piff
+        is expecting this.
+
+        This is an internal method used during the fitting.
+        :param pos:        A numpy array of the u,v positions at which to build 
+                           the model
+        :param C:          A 2D matrix of polynomial coefficients in the form that
+                           the numpy polynomial form is expecting:
+                           p(x,y,c) = sum_{i,j} c_{ij} x^i y^j
+                           
+        :returns:          A numpy array of the calculated p_x(x)*p_y(y) where 
+                           the p functions are polynomials.
+
+        """
+        #Take the u and v components (x and y in the tangent plane)
+        #as our interpolants
+        u = pos[0]
+        v = pos[1]
+        #Call the appropriate function to generate the polynomial model.
+        #By default this is numpy.polyval. Note that despite appearances
+        #this is not a method call - function is a normal python attribute
+        #that happens to be a function.
```

Comment:
A minor style point.  I find it hard to read comments that don't have a space after the `#`.  Would you mind adding a space?  (Here and elsewhere in this file.)